### PR TITLE
Add `-S` argument to `env` in scripts/xon.sh

### LINKED ISSUE #2881: pytest fails 8 tests
## xonfig
```markdown
+------------------+-----------------+
| xonsh            | 0.8.0           |
| Python           | 3.6.5           |
| PLY              | 3.11            |
| have readline    | True            |
| prompt toolkit   | 1.0.15          |
| shell type       | prompt_toolkit1 |
| pygments         | 2.2.0           |
| on posix         | True            |
| on linux         | True            |
| distro           | Fedora          |
| on darwin        | False           |
| on windows       | False           |
| on cygwin        | False           |
| on msys2         | False           |
| is superuser     | False           |
| default encoding | utf-8           |
| xonsh encoding   | utf-8           |
| encoding errors  | surrogateescape |
+------------------+-----------------+
```

## Expected Behavior
The tests should pass, I believe.

## Current Behavior
pytest fails with the following tests:

* test_unmodified
* test_crud
* test_crud_path
* test_path
* test_sourcefile
* test_printname
* test_printfile
* test_activate

## Steps to Reproduce
```bash
$ cd xonsh
$ python3 -m pytest
```

## More info
```bash
$ python3 -m xonsh --version
xonsh/0.8.0

$ python3 -m pytest --version
This is pytest version 3.8.2, imported from /home/erez/.local/lib/python3.6/site-packages/pytest.py
setuptools registered plugins:
  xonsh-0.8.0 at /home/erez/xonsh/xonsh/pytest_plugin.py
  pytest-timeout-1.3.2 at /home/erez/.local/lib/python3.6/site-packages/pytest_timeout.py
  pytest-flake8-1.0.2 at /home/erez/.local/lib/python3.6/site-packages/pytest_flake8.py
  pytest-cov-2.6.0 at /home/erez/.local/lib/python3.6/site-packages/pytest_cov/plugin.py
```

### PR DESCRIPTION
**Pull request by carmenbianca**
I ran into #2881 while packaging this for Fedora. This is a super simple fix.

https://bugzilla.redhat.com/show_bug.cgi?id=1695139

Fixes #2881.

### CODE REVIEW

**Comment by carmenbianca** [comment]
Mhm, apparently not all implementations of env have the -S option. This file could also be fixed by running something like:

    exec /usr/bin/env PYTHONUNBUFFERED=1 /usr/bin/xonsh "$@"

But I'm half-certain that that defeats the purpose of that file.

**Comment by laloch** [comment]
The whole line looks somewhat fishy to me.
- `PYTHONUNBUFFERED=1` and `-u` argument do the same thing
- there's no reason to call `/usr/bin/env` at all - the `exec` command will resolve the path to the `python3` binary itself

We could IMO replace the line with
```
exec python3 -u -m xonsh "$@"
```

File: news/env-dash-S.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+**Fixed:**
+
+* An error in the 'xon.sh' executable that only popped up during testing has
+  been fixed.  Specifically: It now calls '/usr/bin/env' with the '-S' option.
```
Comment (laloch) [inline news/env-dash-S.rst]: It now doesn't call `env` at all ;)

Comment (carmenbianca) [reply news/env-dash-S.rst]: You're absolutely right :p Committed a fix for that as well.

Comment (laloch) [reply news/env-dash-S.rst]: Thank you.

**Comment by scopatz** [comment]
merged from master

**Comment by scopatz** [comment]
Thanks @carmenbianca!

**Comment by scopatz** [comment]
We'll cut a new release with this soon

**Pull request merged**